zeroconf/__init__.py

```diff
@@ -1891,6 +1891,7 @@ def request(self, zc: 'Zeroconf', timeout: float) -> bool:
         now = current_time_millis()
         delay = _LISTENER_TIME
         next_ = now + delay
+        first = next_
```

Comment:
This will mean we will reject answers received within 200ms, how about `first = now` instead?
Also, `first` doesn't seem very obvious, maybe change to `request_start` or `first_question` or something like that?

Reply:
@emontnemery I was a bit confused about 
https://github.com/jstasiak/python-zeroconf/blob/19e33a6829846008b50f408c77ac3e8e73176529/zeroconf/__init__.py#L1913

because if you declare: 
https://github.com/jstasiak/python-zeroconf/blob/19e33a6829846008b50f408c77ac3e8e73176529/zeroconf/__init__.py#L1893

on the first loop the if statement would be false, because next_ > now, so the whole DNSOutgoing would be skipped, or do I make a mitake?

Reply:
Oops, I didn't notice that.
That almost looks like a bug as it causes zeroconf to wait 200ms before sending the 1st query?
@jstasiak should the first assignment of `next_` be `next_ = now` or is there a reason to wait?

Reply:
Also I'm wondering why the delay is doubled after each loop?

https://github.com/jstasiak/python-zeroconf/blob/19e33a6829846008b50f408c77ac3e8e73176529/zeroconf/__init__.py#L1930

Reply:
The doubling is to prevent flooding the network with questions to which there will never be an answer, but there should be some reasonable cap to the maximum delay.

Reply:
Huh, yes, it looks like that first unnecessary wait is a bug. Multiplying by two is to get exponential backoff to achieve what @emontnemery says.

Reply:
I'm not sure about cache entry age checking, I'll have to read some code, can't do that today though.